### PR TITLE
Dgassaway/dpt 15508/grpc limits

### PR DESCRIPTION
**Pull request by dgdonut**
* Bump grpc connection limits on CAS server by ~5x, and remove stream limit (probably doens't matter)
* Changed Resolver interface's ResolveAll -> ResolveMany, which supports capping the number of returned addresses (passing 0 behaves like ResolveAll).
* Cap number of resolved addresses passed to fs_util calls at 10 to try and cap total connections

### CODE REVIEW

File: snapshot/bazel/constants.go [hunk 1]
```diff
@@ -22,4 +22,6 @@ const (
 	fsUtilCmdGlobWildCard = "**"
 
 	emptyID = bazel.SnapshotIDPrefix + "-" + bazel.EmptySha + "-0"
+
+	maxResolveToFSUtil = 5
```
Comment (ryancouto) [inline snapshot/bazel/constants.go:28]: Would be helpful to add a comment here detailing why this value exists

File: common/dialer/resolver.go [hunk 2]
```diff
@@ -10,9 +10,10 @@ type Resolver interface {
 	// Resolve resolves a service, getting an address or URL
 	// If a Resolve() call completes successfully but finds no addresses, it will return ("", nil)
 	Resolve() (string, error)
-	// ResolveAll resolves a slice of all known addresses or URLs
-	// If a ResolveAll() call completes successfully but finds no addresses, it will return ([]string{}, nil)
-	ResolveAll() ([]string, error)
+	// ResolveMany resolves a slice of random addresses or URLs
+	// The int parameter specifies the maximum number of addresses to return. If 0, all addresses are returned.
```
Comment (ryancouto) [inline common/dialer/resolver.go]: Given zero values, this interface makes sense, but it still seems awkward... -1 for "return all" might make more sense.

**Review by ryancouto — APPROVED** [review]

File: common/dialer/resolver.go (hunk 2 above)
Comment (ryancouto) [reply common/dialer/resolver.go]: Not blocking on this and don't really have a strong opinion, just thinking out loud

Comment (dgdonut) [reply common/dialer/resolver.go]: fair enough - will make it <= 0 to effectively return all (since there's no valid reason to call with 0 anyway)

**Pull request merged**